cmd/av/commit_split.go

```diff
@@ -0,0 +1,142 @@
+			return err
+		}
+		return nil
+	},
+}
+
+func splitCommit(repo *git.Repo, currentBranchName, currentCommitOID string) error {
+	if _, err := repo.Git("switch", "--detach", currentCommitOID); err != nil {
+		return err
+	}
+	if _, err := repo.Git("reset", "--mixed", "HEAD~"); err != nil {
+		return err
+	}
+	if _, err := repo.Git("add", "--intent-to-add", repo.Dir()); err != nil {
+		return err
+	}
+
+	for {
+		if clean, err := repo.CheckCleanWorkdir(); err != nil {
+			return err
+		} else if clean {
+			break
+		}
+
+		if _, err := repo.Run(&git.RunOpts{
+			Args:        []string{"add", "--patch"},
+			ExitError:   true,
+			Interactive: true,
+		}); err != nil {
+			return err
+		}
+
+		if hasStagedChange, err := repo.HasChangesToBeCommitted(); err != nil {
+			return err
+		} else if !hasStagedChange {
+			return errors.New("nothing is selected to commit")
+		}
+
+		if _, err := repo.Run(&git.RunOpts{
+			// Add --verbose to show the diffs to be committed.
+			Args:        []string{"commit", "--verbose", "--reedit-message", currentCommitOID},
+			ExitError:   true,
+			Interactive: true,
+		}); err != nil {
+			return err
+		}
+	}
+
+	if currentBranchName != "" {
+		newCommitOID, err := repo.RevParse(&git.RevParse{Rev: "HEAD"})
+		if err != nil {
+			return errors.Errorf("cannot get the resulting commit object: %v", err)
+		}
+		if err := repo.UpdateRef(&git.UpdateRef{
+			Ref: "refs/heads/" + currentBranchName,
+			Old: currentCommitOID,
+			New: newCommitOID,
+			// Add this change to reflog so that the user can revert back to the
+			// original state.
+			CreateReflog: true,
+		}); err != nil {
+			return errors.Errorf("cannot update the branch %s to the new commit %s: %v", currentBranchName, newCommitOID, err)
+		}
+		// At this point, the HEAD is still a detached HEAD. Check out the branch.
+		// repo.CheckoutBranch errors out if the repository is at the detached head.
+		// We have to run git checkout in other ways.
+		if _, err := repo.Git("switch", currentBranchName); err != nil {
+			return errors.Errorf("cannot switch to the original branch: %v", err)
+		}
+
+		// TODO: We should rebase the stacks after split.
+	}
+
+	return nil
+}
+
+func commitSplitAbortMessage(branchName, commitOID string) {
+	if branchName == "" {
+		// We started from a detached HEAD.
+		_, _ = fmt.Fprint(
+			os.Stderr,
+			colors.Failure("===================================================="), "\n",
+			colors.Failure("RECOVERY INSTRUCTION"), "\n",
+			"The commit split command is aborted. The HEAD is moved to a different commit than the original commit ",
+			colors.UserInput(commitOID), "\n",
+			"\n",
+			"In order to get back to ", colors.UserInput(commitOID), ",\n",
+			"you need to run ", colors.CliCmd("git switch --discard-changes --detach", commitOID), "\n",
+			colors.Failure("===================================================="), "\n",
+		)
+		return
+	}
+	_, _ = fmt.Fprint(
+		os.Stderr,
```

Comment:
I think `RECOVERY INSTRUCTION` is a bit too severe of a message here. Makes people think they're in a dangerous state and they could lose their work.

Maybe something like
```
====================================================
DETACHED HEAD

The commit split command aborted.
Your original branch {branch} was not modified.
Your Git repository is now in a detached HEAD state.
To revert your changes and switch to your original (unmodified) branch, run:

    git switch --discard-changes {branch}
====================================================

Reply:
Done.